lib/client/cluster_client.go

```diff
@@ -312,6 +284,27 @@ func performMFACeremony(ctx context.Context, clt *ClusterClient, params ReissueP
 	if mfaChal == nil {
 		return nil, trace.BadParameter("server sent a %T on GenerateUserSingleUseCerts, expected MFAChallenge", resp.Response)
 	}
+
+	switch mfaChal.MFARequired {
+	case proto.MFARequired_MFA_REQUIRED_NO:
+		return nil, trace.Wrap(services.ErrSessionMFANotRequired)
+	case proto.MFARequired_MFA_REQUIRED_UNSPECIFIED:
+		if params.MFACheck != nil && !params.MFACheck.Required {
+			return nil, trace.Wrap(services.ErrSessionMFANotRequired)
+		}
+
+		check, err := clt.AuthClient.IsMFARequired(ctx, params.isMFARequiredRequest(clt.tc.HostLogin))
+		if err != nil {
+			return nil, trace.Wrap(MFARequiredUnknown(err))
+		}
+		if !check.Required {
+			return nil, trace.Wrap(services.ErrSessionMFANotRequired)
+		}
+
+		params.MFACheck = check
```

Comment:
Why is it relevant to set this here? Should we be setting it in the other cases too?

Alternatively, could we deprecate/remove this field, or is that detrimental?